File: examples/android/app/src/main/java/com/jnibridge/modules/dom/IUINode.java
package com.jnibridge.modules.dom;

import android.view.View;

import com.jnibridge.EngineScope;

import java.util.ArrayList;

public interface IUINode {
    public EngineScope getScope();

    public int getUniqueId();

    public String getTagName();

    public ArrayList<IUINode> getChildNodes();

    public String getAttribute(String key);

    public void setStyle(String key, String value);

    public void resetStyle();

    public void setAttribute(String key, String value);

    public void removeAttribute(String key);

    public IUINode appendChild(IUINode newChild);

    public boolean contains(IUINode child);

    public IUINode insertBefore(IUINode newChild, IUINode refChild);

    public IUINode removeChild(IUINode child);

    public IUINode replaceChild(IUINode newChild, IUINode oldChild);
}

File: examples/ios/ioscplus/ioscplus/IUINode.swift
//
//  IUINode.swift
//  ioscplus
//
//  Created by song on 2021/8/24.
//

import Foundation

public class IUINode {
    private var _tagName: String;
    private var _uniqueId: Int32;
    private var _scope: JSEngineScope;
    public init(tagName: String, uniqueId: Int32, scope: JSEngineScope) {
        _tagName = tagName
        _uniqueId = uniqueId
        _scope = scope
    }
    
    public func getUniqueId() -> Int32 {
        return _uniqueId;
    }
    
    public func getScope() -> JSEngineScope {
        return _scope
    }

    public func getTagName() -> String {
        return _tagName;
    }
    
    public func getChildNodes() -> [IUINode] {
        return [];
    }

    public func getAttribute(key: String) -> String {
        return ""
    }

    public func setStyle(key: String, value: String) {
        
    }

    public func resetStyle() {
        
    }

    public func setAttribute(key: String, value: String) {
        
    }

    public func removeAttribute(key: String) {
        
    }

    public func appendChild(newChild: IUINode) -> IUINode {
        return newChild
    }

    public func contains(child: IUINode) -> Bool {
        return false
    }

    public func insertBefore(newChild: IUINode, refChild: IUINode?) -> IUINode {
        return newChild
    }

    public func removeChild(child: IUINode) -> IUINode{
        return child
    }

    public func replaceChild(newChild: IUINode, oldChild: IUINode?) -> IUINode {
        return newChild
    }
}
